src/OpenLoco/TownManager.cpp

```diff
@@ -59,6 +63,61 @@ namespace OpenLoco::TownManager
 
     static auto& rawTowns() { return getGameState().towns; }
 
+    // 0x00497348
+    void sub_497348()
+    {
+        for (auto& town : towns())
+        {
+            town.var_38 = 0;
+            town.population = 0;
+            town.var_34 = 0;
+            std::fill(std::begin(town.var_150), std::end(town.var_150), 0);
+        }
+
+        Map::TilePosRangeView tileLoop{ { 1, 1 }, { Map::map_columns - 1, Map::map_rows - 1 } };
+        for (const auto& tilePos : tileLoop)
+        {
+            auto tile = Map::TileManager::get(tilePos);
+            for (auto& element : tile)
+            {
+                auto* building = element.as<Map::BuildingElement>();
+                if (building == nullptr)
+                    continue;
+
+                if (building->isGhost())
+                    continue;
+
+                if (building->has_40())
+                    continue;
+
+                if (building->multiTileIndex() != 0)
+                    continue;
+
+                auto objectId = building->objectId();
+                auto* buildingObj = ObjectManager::get<BuildingObject>(objectId);
+                auto producedQuantity = buildingObj->producedQuantity[0];
+                uint32_t population;
+                if (!building->isConstructed())
+                {
+                    population = 0;
+                }
+                else
+                {
+                    population = producedQuantity;
+                }
+                auto* town = sub_497DC1(tilePos, population, producedQuantity, 0, 1);
```

Comment:
this feels like the second variable is for the capacity of the building which would mean that `town->var_34` is the `populationCapacity`

Reply:
Renamed. `producedQuantity` still seems sort of unrelated or weirdly named in this context.

Reply:
The produced quantity makes sense when you think that buildings first produced item is people. So the produced quantity is the number of people produced by the building

Reply:
That makes sense!